dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -674,7 +675,7 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                     }
                 }
                 if let &ty::FnDef(def_id, _) = func.ty(self.body, self.tcx).kind() {
-                    if destination.is_some() {
+                    if true {
```

Comment:
This logic can now execute unconditionally, but I've left the `if true` to have a place to hang a discussion off of because I don't know anything about the cases where `destination` was previously `None` and would like to get some eyes on this bit of delta. When would `destination` have been `None`? Do we need to ignore these cases some other way now?

Reply:
Leaving for reference. 
 It's quite a long `if`.

https://github.com/immunant/c2rust/blob/a15c7df884785caa1f88509d1b450a04d2cfd3fa/dynamic_instrumentation/src/instrument_memory.rs#L678-L740